apache-libraries-2/src/main/java/com/baeldung/dynamicrouter/DynamicRouterBean.java

```diff
@@ -0,0 +1,28 @@
+package com.baeldung.dynamicrouter;
+
+import org.apache.camel.ExchangeProperties;
+
+import java.util.Map;
+
+public class DynamicRouterBean {
+    public String route(String body, @ExchangeProperties Map<String, Object> properties) {
+        int invoked = 0;
```

Comment:
Can be simplified to: `int invoked = (int) properties.getOrDefault("invoked", 0) + 1;`

Reply:
Thanks @Maiklins